cni/cni.go

```diff
@@ -346,24 +353,64 @@ func (config *NetworkConfig) GetEndpointInfo(
 }
 
 // GetCurrResult gets the result object
-func GetCurrResult(network *network.NetworkInfo, endpoint *network.EndpointInfo, ifname string) cniTypesCurr.Result {
+func GetCurrResult(network *network.NetworkInfo, endpoint *network.EndpointInfo, ifname string, cniConfig *NetworkConfig) cniTypesCurr.Result {
 	result := cniTypesCurr.Result{
 		IPs:    []*cniTypesCurr.IPConfig{},
 		Routes: []*cniTypes.Route{}}
 
 	var iFace = GetInterface(endpoint)
-	var ip = GetIP(network, endpoint)
-	ip.InterfaceIndex = 0
-
-	cIP := cniTypesCurr.IPConfig{
-		Version: ip.Version,
-		Address: net.IPNet{
-			IP:   ip.Address.IP,
-			Mask: ip.Address.Mask},
-		Gateway:   ip.Gateway,
-		Interface: &ip.InterfaceIndex,
-	}
-	result.IPs = append(result.IPs, &cIP)
+
+	if cniConfig.OptionalFlags.EnableDualStack == false {
+
+		var ip = GetIP(network, endpoint)
+		ip.InterfaceIndex = 0
+			
+    	cIP := cniTypesCurr.IPConfig{
+	    	Version: ip.Version,
+		    Address: net.IPNet{
+			    IP:   ip.Address.IP,
+    			Mask: ip.Address.Mask},
+	    	Gateway:   ip.Gateway,
+		    Interface: &ip.InterfaceIndex,
+	    }
+	    result.IPs = append(result.IPs, &cIP)
+
+    } else {
+
+    	ip4, ip6 := GetDualStackAddresses(endpoint)
```

Comment:
For the long term this could have handled the scenario where HNS supports only ipv4, only ipv6 or both. But you are right and since currently ipv4 address is always present I have updated the code to reflect that (removed the check for ipv4 being nil).